container/go/cmd/create_image_config/createImageConfig.go

```diff
@@ -0,0 +1,413 @@
+}
+
+// mapToKeyValue reverses a map to a '='-separated array of strings in {key}={value} format
+func mapToKeyValue(kvMap map[string]string) []string {
+	keyVals := []string{}
+	concatenated := ""
+	for k, v := range kvMap {
+		concatenated = k + "=" + v
+		keyVals = append(keyVals, concatenated)
+	}
+	return keyVals
+}
+
+// resolveVariables resolves the environment variables embedded in the given value using
+// provided map of environment variable expansions.
+// It handles formats like "PATH=$PATH:..."
+func resolveVariables(value string, environment map[string]string) (string, error) {
+	var i interface{}
+	i = "this is a string type"
+	var errorMet = false
+	mapper := func(p string) string {
+		switch i.(type) {
+		case string:
+			return environment[p]
+		}
+		errorMet = true
+		return ""
+	}
+	if errorMet {
+		err := errors.New("environment variable sought after does not exist in evnrionemnet map")
+		if err != nil {
+			errors.Wrap(err, "failed to create new error")
+		}
+		return "", err
+	}
+	return os.Expand(value, mapper), nil
+}
+
+func main() {
+	flag.Var(&labelsArray, "labels", "Augment the Label of the previous layer.")
+	flag.Var(&ports, "ports", "Augment the ExposedPorts of the previous layer.")
+	flag.Var(&volumes, "volumes", "Augment the Volumes of the previous layer.")
+	flag.Var(&entrypointPrefix, "entrypointPrefix", "Prefix the Entrypoint with the specified arguments.")
+	flag.Var(&env, "env", "Augment the Env of the previous layer.")
+	flag.Var(&command, "command", "Override the Cmd of the previous layer.")
+	flag.Var(&entrypoint, "entrypoint", "Override the Entrypoint of the previous layer.")
+	flag.Var(&layer, "layer", "Layer sha256 hashes that make up this image.")
+	flag.Var(&stampInfoFile, "stampInfoFile", "A list of files from which to read substitutions to make in the provided fields.")
+	flag.Parse()
+	log.Println("Running the Image Config manipulator...")
+
+	if *outputConfig == "" {
+		log.Fatalln("Required option -outputConfig was not specified.")
+	}
+	if *nullEntryPoint == "true" {
+		entrypoint = nil
+	}
+	if *nullCmd == "true" {
+		command = nil
+	}
+
+	// read config file into struct.
+	configPath, err := os.Open(*baseConfig)
+	if err != nil {
+		log.Fatalf("Failed to read the parent image's config file: %v", err)
+	}
+	configFile, err := v1.ParseConfigFile(configPath)
+	if err != nil {
+		log.Fatalf("Failed to successfully parse config file json contents: %v", err)
+	}
+
+	// read manifest file into struct if provided.
+	if *baseManifest != "" {
+		manifestPath, err := ioutil.ReadFile(*baseManifest)
+		if err != nil {
+			log.Fatalf("Failed to read the parent image's manifest: %v", err)
+		}
+		manifestFile := v1.Manifest{}
+		if err = json.Unmarshal([]byte(manifestPath), &manifestFile); err != nil {
+			log.Fatalf("Failed to successfully read manifest file contents: %v", err)
+		}
+	}
+
+	var createTime string
+	if *creationTime == "" {
+		creationTime = nil
```

Comment:
Avoid modifying parsed flag variables. This repurposes them and can be confusing when their values differ from the provided command line arguments later.